geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -187,6 +188,224 @@ object GeoMesaSpark extends LazyLogging {
+      // Skip the id
+      if (index != 0) {
+        featureAttributes(index) match {
+          case attr if attr != null && attr.getClass == classOf[Integer] =>
+            Some(index, "Integer")
+          case attr if attr != null && attr.getClass == classOf[java.lang.Long] =>
+            Some(index, "Long")
+          case attr if attr != null && attr.getClass == classOf[java.lang.Double] =>
+            Some(index, "Double")
+          case _ => None
+        }
+      } else {
+        None
+      }
+    }).toArray
+    val countable = sc.broadcast(countableIndices)
+
+    // Create a Simple Feature Type based on what can be aggregated
+    val sftBuilder = new SftBuilder()
+    sftBuilder.stringType(key)
+    sftBuilder.multiPolygon("geom")
+    sftBuilder.intType("count")
+    val featureProperties = data.first.getProperties.toSeq
+    countableIndices.foreach{ case (index, clazz) =>
+        val featureName = featureProperties.apply(index).getName
+        clazz match {
+          case "Integer" => sftBuilder.intType("total_" + featureName)
+          case "Long" => sftBuilder.longType("total_" + featureName)
+          case "Double" => sftBuilder.doubleType("total_" + featureName)
+        }
+        sftBuilder.doubleType("avg_"+featureProperties.apply(index).getName)
+    }
+    val coverSft = SimpleFeatureTypes.createType("aggregate",sftBuilder.getSpec)
+
+    // Register it with kryo and send it to executors
+    GeoMesaSpark.register(Seq(coverSft))
+    val newSfts = sc.broadcast(GeoMesaSparkKryoRegistrator.typeCache.values.map{ sft =>
+      (sft.getTypeName, SimpleFeatureTypes.encodeType(sft))
+    }.toArray)
+    keyedData.foreachPartition{ iter =>
+      newSfts.value.foreach{ case (name, spec) =>
+        val newSft = SimpleFeatureTypes.createType(name, spec)
+        GeoMesaSparkKryoRegistrator.putTypeIfAbsent(newSft)
+      }
+    }
+
+    // Reduce features by their covering area
+    val aggregate = reduceAndAggregate(keyedData, countable)
+
+    // Send a map of cover name -> geom to the executors
+    import org.locationtech.geomesa.utils.geotools.Conversions._
+    val coverMap: scala.collection.Map[String, Geometry] =
+      coveringSet.map{ sf =>
+        sf.getAttribute(key).asInstanceOf[String] -> sf.geometry
+      }.collectAsMap
+
+    val broadcastedCoverMap = sc.broadcast(coverMap)
+
+    // Compute averages and set cover names and geometries
+    aggregate.mapPartitions{ iter =>
+      import org.locationtech.geomesa.utils.geotools.Conversions.RichSimpleFeature
+
+      iter.flatMap{ case (coverName, sf) =>
+        if (sf.getType.getTypeName == "aggregate") {
+          sf.getProperties.foreach{ prop =>
+            val name = prop.getName.toString
+            if (name.startsWith("total_")) {
+              val count = sf.get[Integer]("count")
+              val avg = prop.getValue match {
+                case (a: Integer) => a / count
```

Comment:
Since I'm skipping over features that aren't of type "aggregate", no. 
When I create features of that type, `count` is initialized to two, and when features of that type are merged, `count` only increases. The only case I could see is some sort of overflow... I could add a check for it being 0 or negative, but then what would I use instead?